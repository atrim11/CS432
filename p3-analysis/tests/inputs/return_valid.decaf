def bool anotherFunction() {
    return false; // No error expected
}

def int main() {
    int x;
    x = anotherFunction();
    return 0;
}