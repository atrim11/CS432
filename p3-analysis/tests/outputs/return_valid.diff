--- outputs/return_valid.txt	2024-10-23 13:25:37.698275514 -0400
+++ expected/return_valid.txt	2024-10-22 14:35:43.655120886 -0400
@@ -1,7 +1,3 @@
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
-ERROR: No 'type' attribute
 Program [line 1]
 SYM TABLE:
  print_int : (int) -> void
