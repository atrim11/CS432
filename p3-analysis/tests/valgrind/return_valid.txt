==2071702== Memcheck, a memory error detector
==2071702== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==2071702== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==2071702== Command: ../decaf inputs/return_valid.decaf
==2071702== 
Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 anotherFunction : () -> bool
 main : () -> int

  FuncDecl name="anotherFunction" return_type=bool parameters={} [line 1]
  SYM TABLE:

    Block [line 1]
    SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 5]
  SYM TABLE:

    Block [line 5]
    SYM TABLE:
     x : int

==2071702== 
==2071702== HEAP SUMMARY:
==2071702==     in use at exit: 0 bytes in 0 blocks
==2071702==   total heap usage: 1,920 allocs, 1,920 frees, 240,028 bytes allocated
==2071702== 
==2071702== All heap blocks were freed -- no leaks are possible
==2071702== 
==2071702== For lists of detected and suppressed errors, rerun with: -s
==2071702== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
