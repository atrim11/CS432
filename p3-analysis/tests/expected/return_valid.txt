Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 anotherFunction : () -> bool
 main : () -> int

  FuncDecl name="anotherFunction" return_type=bool parameters={} [line 1]
  SYM TABLE:

    Block [line 1]
    SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 5]
  SYM TABLE:

    Block [line 5]
    SYM TABLE:
     x : int

